galaxy_ng/app/access_control/statements/roles.py

```diff
@@ -1,67 +1,110 @@
+            "galaxy.add_containerregistryremote",
+            "galaxy.change_containerregistryremote",
+            "galaxy.delete_containerregistryremote",
+        ],
+        "description": (
+            "Push, delete, and change execution environments. "
+            "Create, delete and change remote registries.")
     },
-    "NamespaceViewSet": {
-        "LOCKED_ROLES": {
-            "galaxy.content_admin": [
-                "ansible.modify_ansible_repo_content",
-            ],
-            "galaxy.namespace_owner": [
-                "galaxy.add_namespace",
-                "galaxy.change_namespace",
-                "galaxy.delete_namespace",
-                "galaxy.view_namespace",
-                "galaxy.upload_to_namespace",
-                "ansible.delete_collection",
-            ],
-            "galaxy.publisher": [
-                "galaxy.upload_to_namespace",
-                "ansible.delete_collection",
-            ],
-            "galaxy.group_admin": [
-                "galaxy.view_group",
-                "galaxy.delete_group",
-                "galaxy.add_group",
-                "galaxy.change_group",
-            ],
-            "galaxy.user_admin": [
-                "galaxy.view_user",
-                "galaxy.delete_user",
-                "galaxy.add_user",
-                "galaxy.change_user",
-            ],
-        },
+    "galaxy.execution_environment_namespace_owner": {
+        "permissions": [
+            "container.change_containernamespace",
+            "container.namespace_push_containerdistribution",
+            "container.namespace_change_containerdistribution",
+            "container.namespace_modify_content_containerpushrepository",
+            "container.namespace_add_containerdistribution",
+        ],
+        "description": (
+            "Create and update execution environments under existing "
+            "container namespaces.")
     },
-    "SyncListViewSet": {
-        "LOCKED_ROLES": {
-            "galaxy.synclist_owner": [
-                "galaxy.add_synclist",
-                "galaxy.change_synclist",
-                "galaxy.delete_synclist",
-                "galaxy.view_synclist",
-                "ansible.change_collectionremote",
-            ],
-        }
+    "galaxy.execution_environment_collaborator": {
+        "permissions": [
+            "container.namespace_push_containerdistribution",
+            "container.namespace_change_containerdistribution",
+            "container.namespace_modify_content_containerpushrepository",
+        ],
+        "description": "Change existing execution environments."
+    },
+    "galaxy.content_admin": {
+        "permissions": [
+            "ansible.modify_ansible_repo_content",
+        ],
+        "description": "Approve and reject collections."
+    },
+    "galaxy.namespace_owner": {
+        "permissions": [
+            "galaxy.change_namespace",
+            "galaxy.upload_to_namespace",
```

Comment:
This seems like a good change to limit the namespace owner to uploads and editing the namespace object (not adding/delete namespaces nor deleting collections).

What is the clearest way to see that the role `galaxy.namespace_owner` and it's permissions will apply to a namespace object, and not to a namespace model (this role previously had `galaxy.add_namespace` permission which only makes sense at the model level)

Reply:
@awcrosby there isn't a distinction between object level roles and model level roles. Both can be applied in either setting, which sort of complicates a lot of things. This one is intended to be used for objects, but there's not really a way to denote that.

Reply:
So when this migration looks at an existing group and its permissions, what happens when a group has a mix of object and group level permissions... I like the docstring example in `migrate_group_permissions_to_roles` maybe it could get a couple more examples?

    Example:
    If a group has permissions:
        - galaxy.change_namespace (at object level)
        - galaxy.upload_to_namespace (at object level)
        - galaxy.delete_collection (at model level)
        - galaxy.view_group (at model level)
        - galaxy.view_user (at object level)
        
    The following roles would get applied:
        - galaxy.namespace_owner (does it have the same perms each at its previous object/model level?)
        - _permission:galaxy.view_group (at model level?)
        - _permission:galaxy.view_user (at object level?)

Reply:
`OBJECT_PERMISSION_TRANSLATOR` gets applied to object perms and `GLOBAL_PERMISSION_TRANSLATOR` gets applied to model perms

Reply:
For future reference: As long as a role has any permissions for an object you can assign it to that object. The extra permissions are ignored. So if I have a role with galaxy.change_namespace and galaxy.view_user and apply it to a collection namespace, view_user won't do anything.